AEPCore/Sources/eventhub/EventHub.swift

```diff
@@ -195,25 +201,27 @@ final class EventHub {
     ///   - barrier: If true, the `EventHub` will only return `.set` if `extensionName` has moved past `event`
     /// - Returns: The `SharedState` data and status for the extension with `extensionName`
     func getSharedState(extensionName: String, event: Event?, barrier: Bool = true) -> SharedStateResult? {
-        guard let container = registeredExtensions.first(where: { $1.sharedStateName == extensionName })?.value, let sharedState = container.sharedState else {
-            Log.warning(label: LOG_TAG, "Unable to retrieve shared state for \(extensionName). No such extension is registered.")
-            return nil
-        }
+        return eventHubQueue.sync {
```

Comment:
a little bit concerned about the performance implication but we can see if it will cause any problem